tests/test_synced_collection.py

```diff
@@ -551,33 +585,35 @@ def test_repr(self, synced_list):
     def test_str(self, synced_list):
         str(synced_list) == str(synced_list.to_base())
 
-    def test_nested_list(self, synced_list):
-        synced_list.reset([1, 2, 3])
-        synced_list.append([2, 4])
-        child1 = synced_list[3]
-        child2 = synced_list[3]
+    @given(st.lists(JSONDataStrategy, max_size=5), JSONDataStrategy)
+    def test_nested_list(self, synced_list, testdata_list, testdata):
+        synced_list.reset([testdata_list])
+        child1 = synced_list[0]
```

Comment:
This doesn't seem right. In the old code, `child1` is a list: `[2, 4]`. Here, it looks like `child1` is a "data element" (not a list). Carefully review this test.